File: src/view/components/GridBlock.java
package view.components;

import model.structs.Rect;

public class GridBlock extends AbstractDisplayBlock{
    int[][] grid;

    public GridBlock(Rect margin, AlignType verticalAlign, AlignType horizontalAlign, int height, int width) {
        super(margin, verticalAlign, horizontalAlign);
        this.grid = new int[height][width];
    }

    public int[][] getGrid() {
        return grid;
    }

    @Override
    public int getAvailableWidth() {
        return grid[0].length * 2;
    }

    @Override
    public int getAvailableHeight() {
        return grid.length;
    }

    @Override
    protected void updateSize(AbstractDisplayBlock child) {}
}

File: src/view/components/GridBlock.kt
package view.components

import model.structs.Rect

class GridBlock(margin: Rect, verticalAlign: AlignType?, horizontalAlign: AlignType?, height: Int, width: Int) :
    AbstractDisplayBlock(margin, verticalAlign, horizontalAlign) {
    var grid: Array<IntArray> = Array(height) { IntArray(width) }

    override val availableWidth: Int
        get() = grid[0].size * 2

    override val availableHeight: Int
        get() = grid.size

    override fun updateSize(child: AbstractDisplayBlock?) {}
}
